proto/src/determined/experiment/v1/searcher.proto

```diff
@@ -143,9 +143,9 @@ message SetSearcherProgressOperation {
 message CompleteValidateAfterOperation {
   // The ValidateAfterOperation being completed.
   ValidateAfterOperation op = 1;
-  // The value of searcher metric associated with this completed metric.
+  // The value of searcher metric associated with this completed operation.
   // The metric provided should be the metric used to guide HP search.
-  double searcher_metric = 2;
```

Comment:
I have a couple questions here.

First, does this change involve a loss of precision?  It looks to me like the answer is "no, we used to just dump floats right into json and that doesn't change" but I wanted to make sure.  Is it int64 that is string-encoded by grpc-gateway under some circumstances?

Second, are we going to need master-side logic for parsing `"Nan"`, `"Infinity"` and `"-Infinity"` strings?

Reply:
On the first point.

Reply:
hm?

Reply:
On the second point.
I managed to cripple the existing version of the server (the experiments don't render, `sql: Scan error on column index 27, name "best_trial": json: cannot unmarshal string into Go struct field ExperimentTrial.searcher_metric_value of type float64`) by calling `report_completed(float("NaN"))` when using custom searcher. With a regular ASHA searcher nothing good comes out of calling report_completed with NaN, either:
`failed to snapshot experiment, fault tolerance is lost  actor-local-addr="159" actor-system="master" error="failed to snapshot searcher: failed to save search method: failed to save subsearch: json: error calling MarshalJSON for type model.ExtendedFloat64: json: unsupported value: NaN" experiment-id="159" go-type="experiment" job-id="e74647dc-15f2-4b7c-a7f3-c43becfeabf2"`

With this change the searcher gets
```
{"id": 40, "validationCompleted": {"metrics": {"foo": "NaN"}, "requestId": "0863fceb-5c8f-4e43-9a2a-9a4e5cc3ae7f", "validateAfterLength": "500"}}
```
or
```
{"id": 40, "validationCompleted": {"metrics": "NaN", "requestId": "8369738a-9d58-44a0-8def-d3d14924c326", "validateAfterLength": "500"}}
```
which is bad (Custom Searcher is getting back a string) but the database layer does not break.

So,
It is already sort of broken (NaNs don't go over the wire that well). Do we want to fix it? As part of this PR?

Reply:
Oh, and for a "standard" search method we get this with the new code:
`unexpected metric type for ASHA built-in search method NaN`.

Reply:
on the first point, I was confused.  I was thinking how grpc-gateway encodes `uint64`'s as strings.  But doubles are just regular json numbers.  No change there.

On the second point, it's actually really tricky to solve perfectly.  You'd think floats in python should appear as floats in go regardless of value, but due to limitations of the json encoding layer, you'd have to do something like this:
- write a custom encoder that emits any "bad" floats as strings.  This is what `bindings.dump_float()` does, but you'd have to do it recursively on an unspecified struct of values.
- write a custom encoder that reads the resulting `map[string]interface{}`, checking for special strings "NaN", "Infinity", or "-Infinity", and converts those to floats.  This would allow passing special floats but disallow passing those string literals.  Probably worth it, but a lot of work for not a lot of benefit.

So no I don't think you need to worry about it in this PR.  I do think we make tickets for any weird failures the master isn't handling right which exist after this pr lands.  But maybe you actually fixed the bad master-side failures on accident?